analysis_driver/dataset.py

```diff
@@ -451,29 +453,35 @@ def lims_ntf(self):
 
     @property
     def species(self):
+        if self._species is None:
+            self._species = self.sample.get('species_name')
         if self._species is None:
```

Comment:
This if statement is superfluous, as there is an identical one in line 456.

Reply:
yes but line 457 could have changed `self._species`

Reply:
I might be wrong, but if you `get()` a key which does not exist, wouldn't you get a KeyError? So if that would be the case, line 458 would never be reached.

Reply:
Actually I'm reading up and the behaviour I described is when you access a dict directly using keys. So please disregard my comment.